order-service/src/main/java/ru/liga/orderservice/service/OrderService.java

```diff
@@ -64,35 +75,59 @@ public MainOrderListDTO getOrdersByStatus(String status) {
         OrderCreatedDTO orderCreatedDTO = new OrderCreatedDTO();
-        orderCreatedDTO.setId(1);
-        orderCreatedDTO.setSecretPaymentUrl("Some test string");
-        orderCreatedDTO.setEstimatedTimeOfArrival(new Time(10));
+        if(restaurantRepository.findRestaurantById(dto.getRestaurantId()).isPresent()){
+            orderToAdd.setRestaurant(restaurantRepository.findRestaurantById(dto.getRestaurantId()).get());
+            orderToAdd.setStatus(Status.CUSTOMER_CREATED.toString());
+            orderToAdd.setCustomerId(CUSTOMER_ID_MOCK); //заглушка пока не будет авторизации и автоматического присвоения id клиента
+            repository.save(orderToAdd);
+            orderToAdd.setItems(orderItemService.addNewOrderItems(dto, orderToAdd));
+            orderCreatedDTO.setId(1);
+            orderCreatedDTO.setSecretPaymentUrl("Some test string");
+            orderCreatedDTO.setEstimatedTimeOfArrival(new Time(10));
+        }
         return orderCreatedDTO;
     }
+    public FullOrderDTO setOrderStatus(long id, Status status) {
+        Order orderToChange = checkIfOrderExist(id);
+        orderToChange.setStatus(status.toString());
+        String address = orderToChange.getRestaurant().getAddress();
+        repository.save(orderToChange);
+        FullOrderDTO dtoToGive = new FullOrderDTO();
+        dtoToGive.setItems(mapItemToItemToShowDTO(orderToChange.getItems()));
+        dtoToGive.setRestaurant(orderToChange.getRestaurant());
+        dtoToGive.setTimestamp(orderToChange.getTimestamp());
+        if(status.toString().equals("KITCHEN_PREPARING")) {
+            if(address.contains("Moscow")) {
+                rabbit.sendMessage("Order with id " + id + " waiting for delivery", "delivery.moscow");
+            } else {
+                rabbit.sendMessage("Order with id " + id + " waiting for delivery", "delivery.nizhniy_novgorod");
+            }
+        }
+        return dtoToGive;
+    }
     private static List<ItemToShowCustomerDTO> mapItemToItemToShowDTO(List<OrderItem> itemList) {
         List<ItemToShowCustomerDTO> itemDTOList = new ArrayList<>();
-        for(OrderItem i : itemList) {
+        for(OrderItem item : itemList) {
             ItemToShowCustomerDTO itemDTO = new ItemToShowCustomerDTO();
-            itemDTO.setQuantity(i.getQuantity());
-            itemDTO.setImage(i.getRestaurantMenuItem().getImage());
-            itemDTO.setDescription(i.getRestaurantMenuItem().getDescription());
-            itemDTO.setPrice(i.getPrice());
+            itemDTO.setQuantity(item.getQuantity());
+            itemDTO.setImage(item.getRestaurantMenuItem().getImage());
+            itemDTO.setDescription(item.getRestaurantMenuItem().getDescription());
+            itemDTO.setPrice(item.getPrice());
             itemDTOList.add(itemDTO);
         }
         return itemDTOList;
     }
+    private Order checkIfOrderExist(long id) {
+        if(repository.existsById(id)) {
```

Comment:
` return repository.findOrderById(id).orElseThrow(() -> new NoSuchOrderException("There is no order with id: " + id));`